arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -442,7 +442,12 @@ arangodb::Result fetchRevisions(
       futures.pop_front();
       shoppingLists.pop_front();
       TRI_ASSERT(futures.size() == shoppingLists.size());
-      res = trx.state()->performIntermediateCommitIfRequired(collection.id());
+
+      auto fut =
+          trx.state()->performIntermediateCommitIfRequired(collection.id());
+      TRI_ASSERT(fut.isReady());
```

Comment:
Is this assertion correct?
This would indicate that there is not actually a "Future" here but the above method is synchronous.
Also i do not think this assert is required, with the below `get()` we will wait for the readiness, so not Production issue, may just be triggered within our tests and surprise someone.

If this is correct, please add a comment here why this future is always ready.

Reply:
Yes, this is correct, because this is an intermediate commit on the local rocksdb transaction. `performIntermediateCommitIfRequired` now returns a future, because _iff_ the transaction in question is operates on a replication 2.0 database and is the leader, then the intermediate commit needs to be replicated, which is an asynchronous operation, before we perform the local commit. In all other cases, we synchronously perform the local commit only. Since this is replication1 specific code, everything is synchronous.